unicodetools/src/main/java/org/unicode/text/UCD/VersionedSymbolTable.java

```diff
@@ -0,0 +1,426 @@
+                                        + " with version-suffix dev: the current dev version is "
+                                        + Settings.latestVersion);
+                    }
+                    return Settings.LATEST_VERSION_INFO;
+                } else if (versionNumber.endsWith("α") || versionNumber.endsWith("β")) {
+                    final String versionSuffix =
+                            versionNumber.substring(versionNumber.length() - 1);
+                    versionNumber = versionNumber.substring(0, versionNumber.length() - 1);
+                    if (versionSuffix != Settings.latestVersionPhase.toString()) {
+                        throw new IllegalArgumentException(
+                                "Invalid version-qualifier "
+                                        + versionQualifier
+                                        + " with version-suffix "
+                                        + versionSuffix
+                                        + ": the current stage is "
+                                        + Settings.latestVersionPhase);
+                    }
+                    if (!versionNumber.isEmpty()
+                            && VersionInfo.getInstance(versionNumber)
+                                    != Settings.LATEST_VERSION_INFO) {
+                        throw new IllegalArgumentException(
+                                "Invalid version-qualifier "
+                                        + versionQualifier
+                                        + " with version-suffix "
+                                        + versionNumber
+                                        + ": the current "
+                                        + versionSuffix
+                                        + " version is "
+                                        + Settings.latestVersion);
+                    }
+                    return Settings.LATEST_VERSION_INFO;
+                } else {
+                    var result = VersionInfo.getInstance(versionNumber);
+                    if (result == Settings.LATEST_VERSION_INFO && requireSuffixForLatest) {
+                        throw new IllegalArgumentException(
+                                "Invalid version-qualifier "
+                                        + versionQualifier
+                                        + " version-suffix "
+                                        + Settings.latestVersionPhase
+                                        + " required for unpublished version");
+                    }
+                    return result;
+                }
+            }
+        }
+    }
+
+    private static Map<UcdPropertyValues.General_Category_Values, String[]>
+            COARSE_GENERAL_CATEGORIES =
+                    Map.of(
+                            UcdPropertyValues.General_Category_Values.Other,
+                            new String[] {"Cc", "Cf", "Cn", "Co", "Cs"},
```

Comment:
optional / cleaner: how about changing String[] to a List or a Set, and using `List.of("Cc", "Cf", "Cn", "Co", "Cs")` or Set.of()?